### PR TITLE
Resolve issues with RSpec setup

### PR DESCRIPTION
**Pull request by spicycode**
With this PR all issues with running tests locally are resolved.  Hopefully so for TravisCI as well.

I'll outline the changes I did below.

- Use `rspec-rails` instead of `rspec` for Rails projects. See https://github.com/rspec/rspec-rails#installation for details. 
- Regenerate default rspec install. This generates different output when `rspec-rails` is in the Gemfile vs `rspec`. The command I ran was `rails generate rspec:install`. 
- Require `rails_helper`, not `spec_helper`. At some point in recent RSpec versions `rails_helper` became the default require in test files.  `spec_helper` is required by `rails_helper`.
-  Move to `expect` syntax (`should` syntax is deprecated, but still works).
- Remove extra RSpec rake task require (this was causing duplicate test suite runs).

### CODE REVIEW

File: Rakefile [hunk 1]
```diff
@@ -4,7 +4,3 @@
 require_relative 'config/application'
 
 Rails.application.load_tasks
-
```
Comment (spicycode) [inline Rakefile:7]: Removing these removed the duplicate test runs.  Now the initializer in `rspec-rails` sets up the test suite to run just once.

Comment (rebootunity) [reply Rakefile:7]: Cool - lines 8-10 were from a stack overflow suggestion, I'm glad we don't need them

**Review by rebootunity — COMMENTED** [review]
So many thanks for setting it back on the right track

**Comment by spicycode** [comment]
@fifthbymidwest I'm working through issues with Travis setup now, but that should be the final piece.

**Comment by spicycode** [comment]
fifthbymidwest github io - travis ci 2017-07-07 15-46-22

Everything is passing now.  Pending your review this is good to merge from my end. (The build failing badge is because master is failing).

**Pull request merged**